Diary/src/main/java/com/microsoft/device/display/samples/diary/ui/pages/CalendarPage.kt

```diff
@@ -0,0 +1,125 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shadow
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.viewinterop.AndroidView
+import com.microsoft.device.display.samples.diary.R
+import com.microsoft.device.dualscreen.twopanelayout.TwoPaneScope
+import java.time.LocalDate
+
+/**
+ *  The CalendarView Composable which also contains the Diary Content for the selected day
+ */
+@Composable
+fun TwoPaneScope.CalendarPage(
+    content: String,
+    selectedDate: LocalDate,
+    updateDate: (LocalDate) -> Unit,
+    updateContent: () -> Unit
+) {
+    val twoPaneScope = this
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    Text(text = stringResource(id = R.string.app_name))
+                },
+                contentColor = Color.White,
+                backgroundColor = MaterialTheme.colors.primaryVariant,
+                actions = {
+                    if (twoPaneScope.isSinglePane) {
+                        IconButton(
+                            onClick = { twoPaneScope.navigateToPane2() }
+                        ) {
+                            Icon(
+                                imageVector = Icons.Default.Edit,
+                                contentDescription = "Edit diary"
+                            )
+                        }
+                    }
+                }
+            )
+        }
+    ) {
+
+        Column {
+            AndroidView(
+                {
+                    CalendarView(it)
+                        .apply {
+                            layoutParams = ViewGroup.LayoutParams(
+                                ViewGroup.LayoutParams.MATCH_PARENT,
+                                ViewGroup.LayoutParams.WRAP_CONTENT
+                            )
+                        }
+                },
+                modifier = Modifier.wrapContentWidth(),
+                update = {
+                    it.setOnDateChangeListener { _, year, month, dayOfMonth ->
+                        val currentSelectedDate = LocalDate.of(year, month + 1, dayOfMonth)
+                        updateDate(currentSelectedDate)
+                        updateContent()
```

Comment:
you may also have to call updateText() here or add some other logic - I'm getting a bug where I edit an entry, save, then click another day, and the text from the prevoius entry is still showing in the edit text (instead of "Let's start writing for the day")

Reply:
Sure, I tried doing that, but it somehow worked on my device, probably I am not correctly repeating what you tried, we can meet tomorrow morning and try to fix it, thanks!

Reply:
ah I didn't try updating the code, that was just a suggestion for something that might help! I might just be misunderstanding the flow of the app, I just wasn't sure if the edit text should update to the text from the current day

Reply:
Ah I see, will try out a few manual tests and if I see anything funny, would track that bug :) thanks!